subsys/bluetooth/host/hci_core.c

```diff
@@ -2092,6 +2092,12 @@ static void hci_encrypt_change(struct net_buf *buf)
 		return;
 	}
 
+	if (conn->encrypt == evt->encrypt) {
```

Comment:
Shouldn't this be above https://github.com/nrfconnect/sdk-zephyr/pull/1796/files#diff-774b3de86adb275da54becf2a0958a2d4dae03d63e47840037ecc575e9d658b4R2088 ?

Reply:
If `status` != `0U` then `if (conn->encrypt == evt->encrypt)`  is not needed. Hence, the changes in this commit are corrrect.